### PR TITLE
handle tombstones and undefined manifest props

### PR DESCRIPTION
**Pull request by kuenishi**
addresses #849 .

### CODE REVIEW

**Comment by kuenishi** [comment]
TODO: write tests by using `delete_mode` .

**Comment by reiddraper** [comment]
@kuenishi any preference on a milestone for this issue?

**Comment by kuenishi** [comment]
Now I am investigating #840 with @zschlag, which is in 1.5.0. This is expected to solve #840 but not sure and some weird things left. For the time being I'll mark this as 1.5.0.

**Comment by reiddraper** [comment]
Thanks @kuenishi.

File: src/riak_cs_storage.erl [hunk 1]
```diff
@@ -102,8 +102,9 @@ object_size_map({error, notfound}, _, _) ->
 object_size_map(Object, _, _) ->
     try
         AllManifests = [ binary_to_term(V)
-                         || V <- riak_object:get_values(Object) ],
-        Resolved = riak_cs_manifest_resolution:resolve(AllManifests),
+                         || V <- riak_object:get_values(Object), V /= <<>> ],
```
Comment (shino) [inline src/riak_cs_storage.erl]: How about replacing `/= <<>>` by riak_cs_utils:has_tombstone/1`?

**Comment by shino** [comment]
If we had some warning (or error?) logs at catch block, analysis would be easier.
Is it a good idea  that we add log in this PR?

See : https://github.com/basho/riak_cs/issues/837

**Comment by kuenishi** [comment]
Note: after #848 merged, pushing any more commits without rebasing to develop  or running CI tests will break the plt file in our CI environment. Please rebase to develop, or branch including #848 .

**Comment by kuenishi** [comment]
@shino yeah, logging is good, definitely.

**Comment by shino** [comment]
Do we need some tests for `props=undefined` and/or tombstoned manifest?

**Comment by kuenishi** [comment]
Thanks to @zschlag , this patch will definitely solve #840 . It wasn't `.props=undefined` but old `manifest_v1` . I think adding tests may be another issue. Thoughts?

**Comment by kuenishi** [comment]
Note: Maybe we should change our codes to print stacktrace.

**Comment by kuenishi** [comment]
We saw `function_clause [{proplists,get_value,[multipart,undefined,undefined]` and `reason error:{badrecord,multipart_manifest_v1} [{riak_cs_storage,count_multipart_parts,2,[{file,"src/riak_cs_storage.erl"},{line,160}]}` in the log.

File: src/riak_cs_storage.erl [hunk 2]
```diff
@@ -111,7 +109,14 @@ object_size_map(Object, _, _) ->
             _ ->
                 [{MPparts, MPbytes}]
         end
-    catch _:_ ->
+    catch Type:Reason ->
+            _ = lager:log(error,
+                          self(),
+                          "Riak CS object list map failed for ~p:~p with reason ~p:~p",
```
Comment (shino) [inline src/riak_cs_storage.erl:115]: list?

**Comment by kuenishi** [comment]
This branch will be rebased.

**Pull request closed without merging**